### PR TITLE
Result of fighting in grass is now displayed in grass page instead of home

### LINKED ISSUE #206: Result of fighting with pokemon could have own page
Having it on index looks weird:

https://ctrlv.cz/shots/2016/12/15/HylB.png

### PR DESCRIPTION
**Pull request by Ahimtar**
Resolves #206

### CODE REVIEW

File: spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java [hunk 1]
```diff
@@ -190,7 +190,12 @@ public String fightWild(@RequestParam String mode, ServletRequest r, Model model
         session.setAttribute("authenticatedUser", user);
 
         LOG.debug(user.getUserName() + (mode_ == WildPokemonFightMode.CATCH ? " catching" : " fighting") + " wild " + wildPokemon.getName().getName() + " level " + wildPokemon.getLevel() + " -> " + (fightResult ? "Success!" : "Failed"));
-        return "home";
+        wildPokemon = pokemonFacade.generateWildPokemon(user);
+        session.setAttribute("wildPokemon", wildPokemon);
+        model.addAttribute("wildPokemon", wildPokemon);
+        model.addAttribute("trainersPokemon", user.getPokemons().get(0));
```
Comment (mseleng) [inline spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java]: You should always do an `isEmpty()` check. Just to be sure

File: spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java [hunk 2]
```diff
@@ -190,7 +190,12 @@ public String fightWild(@RequestParam String mode, ServletRequest r, Model model
...
+        LOG.debug("A wild " + wildPokemon.getName().getName() + " level " + wildPokemon.getLevel() + " appeared for " + user.getUserName());
```
Comment (mseleng) [inline spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java]: Please, look at other files on how the parameters should be passed to logs. String concatenation is not necessary here.

File: spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java (hunk 1 above)
Comment (Ahimtar) [inline spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java]: Isn't it covered a few lines earlier in the method, does it need to be done again?

File: spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java (hunk 2 above)
Comment (Ahimtar) [inline spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java]: The whole controller is done in this style, should i change it all for better?

File: spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java (hunk 1 above)
Comment (mseleng) [inline spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java]: Which line number?

File: spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java (hunk 2 above)
Comment (mseleng) [inline spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java]: Yes, please.

File: spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java (hunk 1 above)
Comment (Ahimtar) [inline spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java]: 185-189

File: spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java [hunk 3]
```diff
@@ -72,7 +72,7 @@ public String list(@RequestParam String period, Model model) {
                 break;
         }
         model.addAttribute("fights", fights);
-        LOG.debug("Viewing fights of " + period);
+        LOG.debug("Viewing fights of ", period);
```
Comment (mseleng) [inline spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java]: You forgot parentheses here

File: spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java [hunk 4]
```diff
@@ -111,18 +111,18 @@ public String listFightsOfUser(@RequestParam String username, Model model) {
...
             return "home";
         }
         Collection<FightDTO> fights = fightFacade.listFightsOfTrainer(userFacade.findUserByUserName(username).iterator().next());
         if (fights.isEmpty()) {
             model.addAttribute("alert_warning", "No fights of this user found");
             model.addAttribute("trainer", userFacade.findUserByUserName(username).iterator().next());
-            LOG.debug("Tried to look up fights of '" + username + "' - user doesn't have any fight history.");
-            return "user/detail";
+            LOG.debug("Tried to look up fights of '{}' - user doesn't have any fight history.", username);
+            return "/user/detail";
```
Comment (mseleng) [inline spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java]: Slash here is redundant. Please remove it, to be consistent

File: spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java [hunk 5]
```diff
@@ -263,11 +264,11 @@ public String detail(@PathVariable Long id, Model model) {
         FightDTO fight = fightFacade.findFightById(id);
         if (fight == null) {
             model.addAttribute("alert_warning", "Error: Fight not found in database!");
-            LOG.error("Error: Fight ID " + id + " not found in database!");
+            LOG.error("Error: Fight ID {} not found in database!", id);
             return "home";
         }
         model.addAttribute("fight", fight);
-        LOG.debug("Viewing detail of fight " + id);
+        LOG.debug("Viewing detail of fight ", id);
```
Comment (mseleng) [inline spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java]: Forgotten parentheses here also

**Pull request merged**